typed-racket-lib/typed-racket/typecheck/tc-lambda-unit.rkt

```diff
@@ -196,11 +205,21 @@
 
   (define arg-list (formals-positional formals))
   (define rest-id (formals-rest formals))
+
+  (define eta-expand?
```

Comment:
The name makes it sound like we're trying to decide whether we should eta expand, whereas it looks more like trying to identify whether the code is an eta expansion.